roles/prometheusexporter/templates/deployment.yml.j2

```diff
@@ -130,3 +130,6 @@ spec:
           - key: config.yml
             path: config.yml
 {% endif %}
+{% if node_selector is defined %}
+      nodeSelector: {{ node_selector }}
+{% endif %}
```

Comment:
I have tested it and it does not work, running the operator locally with a modified CR from examples directory (and adding the new `nodeSelector` field), it fails with:
```json
 TASK [Manage Deployment for PrometheusExporter staging-backend-redis on Namespace prom-exporter] ******************************** 
fatal: [localhost]: FAILED! => {"changed": false, "error": 422, "msg": "Failed to patch object: {\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"Deployment.apps \\\"prometheus-exporter-redis-staging-backend-redis\\\" is invalid: [spec.template.spec.nodeSelector: Invalid value: \\\"u'node'\\\": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.template.spec.nodeSelector: Invalid value: \\\"u'test'\\\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]\",\"reason\":\"Invalid\",\"details\":{\"name\":\"prometheus-exporter-redis-staging-backend-redis\",\"group\":\"apps\",\"kind\":\"Deployment\",\"causes\":[{\"reason\":\"FieldValueInvalid\",\"message\":\"Invalid value: \\\"u'node'\\\": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')\",\"field\":\"spec.template.spec.nodeSelector\"},{\"reason\":\"FieldValueInvalid\",\"message\":\"Invalid value: \\\"u'test'\\\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')\",\"field\":\"spec.template.spec.nodeSelector\"}]},\"code\":422}\n", "reason": "Unprocessable Entity", "status": 422}
```

After testing a way to do so, this is the code that works:
```suggestion
{% if node_selector is defined %}
      nodeSelector:
{% for key, value in node_selector.items() %}
        {{ key | safe }}: {{ value | safe }}
{% endfor %}
{% endif %}
```